src/ansi-c/c_typecheck_code.cpp

```diff
@@ -807,10 +809,29 @@ void c_typecheck_baset::typecheck_dowhile(code_dowhilet &code)
   break_is_allowed=old_break_is_allowed;
   continue_is_allowed=old_continue_is_allowed;
 
+  typecheck_spec_loop_assigns(code);
   typecheck_spec_loop_invariant(code);
   typecheck_spec_decreases(code);
 }
 
+void c_typecheck_baset::typecheck_spec_loop_assigns(codet &code)
+{
+  if(code.find(ID_C_spec_loop_assigns).is_not_nil())
+  {
+    for(const exprt &operand : static_cast<exprt &>(code.add(ID_C_spec_loop_assigns)).operands())
+    {
+      if(
+        operand.id() != ID_symbol && operand.id() != ID_ptrmember &&
```

Comment:
These look like the restrictions on other assign clauses.  Is there any chance we could make them common.  Esp. as I recall someone saying that arrays were going to be added.